sdk/go/pulumi/types.go

```diff
@@ -564,14 +564,13 @@ func AllWithContext(ctx context.Context, inputs ...interface{}) ArrayOutput {
 	return ToOutputWithContext(ctx, inputs).(ArrayOutput)
 }
 
-func gatherDependencies(v interface{}) ([]Resource, workGroups) {
+func gatherJoins(v interface{}) workGroups {
```

Comment:
See my other comment for a scenario where this falls down.

In PR #9506 and #9732, I modified `await` to properly implement a monadic `join` operation that accumulated the fields on output state such as `dependencies`, and introduced tests on that PR.

In general we shouldn't be inspecting the dependencies property until we await the output.

Reply:
Awesome! Sounds like good engineering pays off! Thanks for the explanation here, that was very helpful in getting context for these changes.